frontend-project/src/utils/utils.ts

```diff
@@ -1,51 +1,55 @@
 import { parse } from 'querystring';
 import pathRegexp from 'path-to-regexp';
+import { Route } from '@/models/connect';
 
 /* eslint no-useless-escape:0 import/prefer-default-export:0 */
 const reg = /(((^https?:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+(?::\d+)?|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)$/;
-export const isUrl = path => reg.test(path);
-export const isAntDesignPro = () => {
+
+export const isUrl = (path: string): boolean => reg.test(path);
+
+export const isAntDesignPro = (): boolean => {
   if (ANT_DESIGN_PRO_ONLY_DO_NOT_USE_IN_YOUR_PRODUCTION === 'site') {
     return true;
   }
-
   return window.location.hostname === 'preview.pro.ant.design';
-}; // 给官方演示站点用，用于关闭真实开发环境不需要使用的特性
+};
 
-export const isAntDesignProOrDev = () => {
+export const isAntDesignProOrDev = (): boolean => {
   const { NODE_ENV } = process.env;
-
   if (NODE_ENV === 'development') {
     return true;
   }
-
   return isAntDesignPro();
 };
+
 export const getPageQuery = () => parse(window.location.href.split('?')[1]);
-/**
- * props.route.routes
- * @param router [{}]
- * @param pathname string
- */
 
-export const getAuthorityFromRouter = (router = [], pathname) => {
-  const authority = router.find(({ path }) => path && pathRegexp(path).exec(pathname));
+export const getAuthorityFromRouter = <T extends Route>(
+  router: T[] = [],
+  pathname: string,
+): T | undefined => {
+  const authority = router.find(
+    ({ routes, path = '/' }) =>
+      (path && pathRegexp(path).exec(pathname)) ||
```

Comment:
Nie korzystasz przypadkiem z innej wersji Ant Desing Pro? To nie wygląda jakby było tym samym w TypeScript'cie.

Reply:
wygląda mi na to, że tak, po prostu w związku z rekurencyjnym interejsem Route ta funkcja rekurencyjnie pobiera dla niego authority